### ISSUE TITLE
Add focal loss and relevant hyperparameters

### ISSUE THREAD

**Issue by klieret** [GageDeZoort/gnn_tracking#44]

**Comment by klieret** [comment]
It's finished but waiting for #43 so that we can actually switch out the loss functions
